nx/lib/nx/defn/expr.ex

```diff
@@ -251,6 +251,15 @@ defmodule Nx.Defn.Expr do
     attach_token(token, expr)
   end
 
+  @doc false
+  def boundary(type, key, fun, args) do
+    {params, context} = Enum.map_reduce(args, nil, &to_param_expr(&1, :root, &2))
```

Comment:
I was gonna ask if you should be using `:root` here, but I think that since we're always at the start of a given defn here, it's always `:root`, right?